mapper/pojo-base/src/main/java/org/hibernate/search/mapper/pojo/mapping/definition/annotation/impl/AnnotationPojoInjectableBinderCollector.java

```diff
@@ -0,0 +1,251 @@
+		initChildren();
+		this.injectableBinderModels.add( model );
+		return this;
+	}
+
+	public boolean hasContent() {
+		return injectableBinderModels != null && !injectableBinderModels.isEmpty();
+	}
+
+	private void initChildren() {
+		if ( this.injectableBinderModels == null ) {
+			this.injectableBinderModels = new ArrayList<>();
+		}
+	}
+
+	List<InjectableBinderFieldInjector> injector(Class<?> classType) {
+		return injectors.getOrDefault( classType, List.of() );
+	}
+
+
+	interface InjectableBinderAnnotationProcessor<A extends Annotation> {
+		InjectableBinderFieldInjector process(PojoInjectablePropertyModel<?> property, A annotation);
+	}
+
+
+	interface InjectableBinderFieldInjector {
+		// todo: create some context class instead of this list of params...
+		void inject(Object binder, IndexSchemaElement indexSchemaElement, NamedValues params);
+	}
+
+	private static class FullTextFieldInjectableBinderAnnotationProcessor
+			implements InjectableBinderAnnotationProcessor<FullTextField> {
+
+		static final FullTextFieldInjectableBinderAnnotationProcessor INSTANCE =
+				new FullTextFieldInjectableBinderAnnotationProcessor();
+
+		@Override
+		public InjectableBinderFieldInjector process(PojoInjectablePropertyModel<?> property, FullTextField annotation) {
+			String name = annotation.name();
+			if ( name == null || name.isBlank() ) {
+				name = property.name();
+			}
+			return new InjectableBinderFieldInjectorImpl<>( property, name, f -> {
+				StringIndexFieldTypeOptionsStep<?> step = f.asString();
+
+				Projectable projectable = annotation.projectable();
+				if ( !Projectable.DEFAULT.equals( projectable ) ) {
+					step.projectable( projectable );
+				}
+
+				Searchable searchable = annotation.searchable();
+				if ( !Searchable.DEFAULT.equals( searchable ) ) {
+					step.searchable( searchable );
+				}
+
+				if ( !annotation.searchAnalyzer().isEmpty() ) {
+					step.searchAnalyzer( annotation.searchAnalyzer() );
+				}
+
+				Norms norms = annotation.norms();
+				if ( !Norms.DEFAULT.equals( norms ) ) {
+					step.norms( norms );
+				}
+
+				TermVector termVector = annotation.termVector();
+				if ( !TermVector.DEFAULT.equals( termVector ) ) {
+					step.termVector( termVector );
+				}
+				Highlightable[] highlightable = annotation.highlightable();
+				if ( !( highlightable.length == 1 && Highlightable.DEFAULT.equals( highlightable[0] ) ) ) {
+					step.highlightable(
+							highlightable.length == 0 ? Collections.emptyList() : Arrays.asList( highlightable )
+					);
+				}
+
+				return step;
+			} );
+		}
+	}
+
+	private static class KeywordFieldInjectableBinderAnnotationProcessor
+			implements InjectableBinderAnnotationProcessor<KeywordField> {
+
+		static final KeywordFieldInjectableBinderAnnotationProcessor INSTANCE =
+				new KeywordFieldInjectableBinderAnnotationProcessor();
+
+		@Override
+		public InjectableBinderFieldInjector process(PojoInjectablePropertyModel<?> property, KeywordField annotation) {
```

Comment:
Note the fact that you're manipulating a property model here means we won't be able to use this in javac annotation processors -- because I doubt we can implement `PojoPropertyModel` on top of annotation processor APIs. I'd love to be proven wrong though.
The javac annotation processor code that infers the index model from annotations will likely be completely separate...